src/bun.js/ipc.zig

```diff
@@ -156,17 +156,162 @@ pub const IPCData = struct {
     }
 };
 
-/// This type is shared between VirtualMachine and Subprocess for their respective IPC handlers
-///
-/// `Context` must be a struct that implements this interface:
-/// struct {
-///     globalThis: ?*JSGlobalObject,
-///     ipc: IPCData,
-///
-///     fn handleIPCMessage(*Context, DecodedIPCMessage) void
-///     fn handleIPCClose(*Context, Socket) void
-/// }
-pub fn NewIPCHandler(comptime Context: type) type {
+const NamedPipeIPCData = struct {
+    const uv = bun.windows.libuv;
+
+    pipe: uv.uv_pipe_t,
+    incoming: bun.ByteList = .{}, // Maybe we should use IPCBuffer here as well
+    outgoing: IPCBuffer = .{},
+    connected: bool = false,
+    has_written_version: if (Environment.allow_assert) u1 else u0 = 0,
+    connect_req: uv.uv_connect_t = std.mem.zeroes(uv.uv_connect_t),
+    server: uv.uv_pipe_t = std.mem.zeroes(uv.uv_pipe_t),
+    current_payload_len: usize = 0,
+
+    pub fn processSend(this: *NamedPipeIPCData) void {
+        const bytes = this.outgoing.list.slice()[this.outgoing.cursor..];
```

Comment:
When `outgoing.list` resizes on append, this becomes a use-after-free.

Reply:
fixed